envoy/grpc/async_client.h

```diff
@@ -75,12 +75,13 @@ class RawAsyncStream {
    * Register a callback to be called when high/low write buffer watermark events occur on the
    * stream. This callback must persist beyond the lifetime of the stream or be unregistered via
    * removeWatermarkCallbacks. If there's already a watermark callback registered, this method
-   * will ASSERT-fail.
+   * will trigger ENVOY_BUG.
```

Comment:
Ah, I didn't realize the assert was already in there.  honestly it's a good one to have as a bug (the code predated the macro) so thanks for the change :-)